### PR TITLE
ui: increase build max GC space

### PR DESCRIPTION
**Pull request by NicolasT**
Hopefully this reduces CI build flakiness.

### CODE REVIEW

**Comment by NicolasT** [comment]
Ugh.
```
Command '['/entrypoint.sh', 'build']' in image 'metalk8s-ui-builder:latest' returned non-zero exit status 1: b'\n> metalk8s-platform-ui@0.1.0 build /home/node\n> react-app-rewired --max_old_space_size=3072 build\n\nCreating an optimized production build...\nBrowserslist: caniuse-lite is outdated. Please run:\nnpx browserslist@latest --update-db\n\nWhy you should do it regularly:\nhttps://github.com/browserslist/browserslist#browsers-data-updating\nThe build failed because the process exited too early. This probably means the system ran out of memory or someone called `kill -9` on the process.\nnpm ERR! code ELIFECYCLE\nnpm ERR! errno 1\nnpm ERR! metalk8s-platform-ui@0.1.0 build: `react-app-rewired --max_old_space_size=3072 build`\nnpm ERR! Exit status 1\nnpm ERR! \nnpm ERR! Failed at the metalk8s-platform-ui@0.1.0 build script.\nnpm ERR! This is probably not a problem with npm. There is likely additional logging output above.\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /home/node/.npm/_logs/2021-03-30T10_19_58_581Z-debug.log\n':
b'\n> metalk8s-platform-ui@0.1.0 build /home/node\n> react-app-rewired --max_old_space_size=3072 build\n\nCreating an optimized production build...\nBrowserslist: caniuse-lite is outdated. Please run:\nnpx browserslist@latest --update-db\n\nWhy you should do it regularly:\nhttps://github.com/browserslist/browserslist#browsers-data-updating\nThe build failed because the process exited too early. This probably means the system ran out of memory or someone called `kill -9` on the process.\nnpm ERR! code ELIFECYCLE\nnpm ERR! errno 1\nnpm ERR! metalk8s-platform-ui@0.1.0 build: `react-app-rewired --max_old_space_size=3072 build`\nnpm ERR! Exit status 1\nnpm ERR! \nnpm ERR! Failed at the metalk8s-platform-ui@0.1.0 build script.\nnpm ERR! This is probably not a problem with npm. There is likely additional logging output above.\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /home/node/.npm/_logs/2021-03-30T10_19_58_581Z-debug.log\n'
```

File: ui/package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
   "scripts": {
     "analyze": "source-map-explorer 'build/static/js/*.js'",
     "start": "react-app-rewired start",
-    "build": "react-app-rewired build",
+    "build": "react-app-rewired --max_old_space_size=3072 build",
```
Comment (JBWatenbergScality) [inline ui/package.json:43]: I'm thinking maybe a child node process is started without this V8 option, let's try to set this via NODE_OPTIONS env.

```suggestion
    "build": "NODE_OPTIONS='--max-old-space-size=3072' react-app-rewired --max_old_space_size=3072 build",
```

Comment (NicolasT) [reply ui/package.json:43]: I checked (in `ps` output while doing a build), it is being passed through as-is:
```
  11533 ?        Ss     0:00          \_ /bin/bash /entrypoint.sh build
  11560 ?        Sl     0:00              \_ npm
  11571 ?        S      0:00                  \_ sh -c react-app-rewired --max_old_space_size=3072 build
  11572 ?        Sl     0:00                      \_ node /home/node/node_modules/.bin/react-app-rewired --max_old_space_size=3072 build
  11579 ?        Rl     0:12                          \_ node --max_old_space_size=3072 /home/node/node_modules/react-app-rewired/scripts/build.js
```

**Comment by NicolasT** [comment]
Build step succeeded for all of 2.7, 2.8 (with one restart, unrelated to UI build) and 2.9. 2.7 CI failed (beacon test expiration), 2.9 succeeded, 2.8 still running (single-node succeeded).

I'd like to request a bypass-build-status.

**Review by gdemonet — APPROVED** [review]

**Comment by NicolasT** [comment]
@bert-e status

**Pull request merged**